### ISSUE TITLE
rtcp tag calculated wrong with aead / gcm

### ISSUE THREAD

**Issue by ttyridal** [cisco/libsrtp#137]
It seems libsrtp is doing the wrong thing when building the aad  for rtcp packets protected with AEAD_AES_*_GCM. The result is tags not generated in accordance to rfc7714

in particular, in srtp/srtp.c:srtp_protect_rtcp_aead  (and similar in unprotect)
```
*trailer = htonl(SRTCP_E_BIT);
*trailer |= htonl(seq_num);
tseq = htonl(*trailer);  // THIS SEEMS QUESTIONABLE
srtp_cipher_set_aad(stream->rtcp_cipher, (uint8_t*)&tseq,
```

The net effect is that the ESRTCP word is used as little endian when calculating the tag/hash, while inserted (correctly) in the stream as big endian.

I can not find anything in https://tools.ietf.org/html/rfc7714 suggesting this behavior, and it also does not agree with the test vectors given in the rfc.

**Comment by jfigus** [comment]
Unfortunately fixing this will break backwards compatibility with existing releases of libsrtp.  I can think of a couple of ugly implementation options to retain backwards compatibility while still being able to comply with the spec.  Not fixing this will obviously be a problem for interop with non-libsrtp implementations.

**Comment by jfigus** [comment]
This is resolved by da852b0c253f6494bd70ae254808669acea9ea63.  Thank you for reporting this issue.
